src/metabase/driver/druid.clj

```diff
@@ -70,9 +71,13 @@
                                       :settings (:settings query))))
 
 (defn- process-native [{database-id :database, {query :query} :native}]
-  {:pre [(integer? database-id)]}
-  (let-404 [details (sel :one :field [Database :details] :id database-id)]
-    (do-query details query)))
+  {:pre [(integer? database-id) (string? query)]}
+  (let-404 [details (sel :one :field [Database :details], :id database-id)]
+    (let [query (json/parse-string query keyword)]
+      ;; `annotate` happens automatically as part of the QP middleware for MBQL queries but not for native ones.
+      ;; This behavior was originally so we could preserve column order for raw SQL queries.
+      ;; Since Druid results come back as maps for each row there is no order to preserve so we can go ahead and re-use the MBQL-QP annotation code here.
+      (annotate/annotate query (qp/post-process-native query (do-query details query))))))
```

Comment:
can we simply tweak the annotate function so that it always runs against all queries and only affects the queries it is meant to?

having a driver directly call a generic QP function breaks encapsulation, so it would be better if we could just rely on annotate working on this query instead of having to call it directly.